src/main/java/dk/kb/image/TemplateMostUsedColors.java

```diff
@@ -0,0 +1,136 @@
+
+import java.awt.image.BufferedImage;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Template to extract dominant colors from image.
+ * Implementations need to implement a colorspace to work with.
+ * @param <C> represents the datatype that the specific implementation implements.
+ */
+abstract class TemplateMostUsedColors<C> {
+    int pixelCount = 0;
+
+    /**
+     * Template for color analysis. The template defines how the algorithm runs and the subclasses/implementations defines which colorspace to use.
+     * @param x defines how many results that gets returned.
+     * @return top x colors and their percentage of all pixels.
+     */
+    final List<DominantColorDto> getMostUsedColors(BufferedImage img, int x){
+        // Define buckets
+        List<C> buckets = defineBuckets();
+        // Create bucket counter
+        int[] bucketCount = getBucketCount(img, buckets);
+        // Create combined map of buckets and bucket count
+        Map<C, Integer> bucketsWithCount = combineBucketsAndBucketCount(buckets, bucketCount);
+        // Sort the map
+        List<Map.Entry<C, Integer>> sortedList = sortList(bucketsWithCount);
+        // Return results
+        return returnResult(sortedList, x);
+    }
+
+    /**
+     * Defines which color palette to use and in which colorspace it should be represented.
+     * @return the palette as a list of given type.
+     */
+    abstract List<C> defineBuckets();
+
+    /**
+     * Loop through pixels of input image, get color in colorspace defined by subclass for pixel and +1 to bucket closest to pixel color.
+     * @param buckets list of bucket colors, type is defined by subclass.
+     * @return the integer array bucketCounter, which contains the count for each bucket for the input image,
+     */
+    int[] getBucketCount(BufferedImage img, List<C> buckets){
+        // Create bucket counter array
+        int[] bucketCounter = new int[buckets.size()];
+        // Loop over all pixels in image and get RGB color
+        int height = img.getHeight();
+        int width = img.getWidth();
+        for(int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                // Get RGB for each pixel
+                pixelCount ++;
+                int pixelRGB = img.getRGB(x, y);
+                updateBucketCounter(pixelRGB, buckets, bucketCounter);
+            }
+        }
+        return bucketCounter;
+    }
+
+    /**
+     * Method to update bucketCounter in getBucketCount().
+     * @param pixel The current pixels color value in RGB colorspace.
+     * @param buckets List of color buckets of used type in the implementation.
+     * @param bucketCounter Integer array to store count of buckets.
+     */
+    void updateBucketCounter(int pixel, List<C> buckets, int[] bucketCounter) {
+        // Convert RGB pixel value to OK Lab float
+        // Values for checking max
+        int bestColor = 0;
+        double minDistance = 2147483647;
```

Comment:
`Double.MAX_VALUE` is better that the large number as it implicitly explains to the reader what you are doing.